src/gg.regression.unity.bots/Runtime/Scripts/StateRecorder/MouseInputActionObserver.cs

```diff
@@ -222,17 +147,33 @@ public List<MouseInputActionData> FlushInputDataBuffer(bool ignoreLastClick, boo
             return result;
         }
 
-        private List<ObjectStatus> FindObjectsAtPosition(Vector2 position, IEnumerable<ObjectStatus> statefulObjects, out float maxZDepth)
+        public static List<ObjectStatus> FindObjectsAtPosition(Vector2 position, IEnumerable<ObjectStatus> statefulObjects)
```

Comment:
This method now actually works as you would expect and returns properly sorted results based on depth from camera as a click would be processed.